torch/utils/_sympy/functions.py

```diff
@@ -125,7 +126,7 @@ def eval(cls, base, divisor):
 
         try:
             gcd = sympy.gcd(base, divisor)
-            if gcd != 1:
+            if not equal_valued(gcd, 1):
```

Comment:
> It's too late for cherry picks and I don't think we need this for 2.4

@albanD The motivation is to include the fix here. Otherwise, the users will run into an infinite recursion here. The `sympy` related code will break silently after the new version of `sympy` is available.